### PR TITLE
feat(jellyfish-api-core): add masternode isAppliedCustomTransaction RPC

### PR DESCRIPTION
**Pull request by chanakasameera**
What this PR does / why we need it:
Checks that custom transaction was affected on chain

Which issue(s) does this PR fixes?:
Added isAppliedCustomTransaction rpc as part of ongoing #48

### CODE REVIEW

**Review by chee-chyuan — DISMISSED** [review]
LGTM

File: packages/jellyfish-api-core/src/category/masternode.ts [hunk 1]
```diff
@@ -198,6 +198,17 @@ export class Masternode {
...
 
+  /**
+   * Checks that custom transaction was affected on chain
+   *
+   * @param {string} transactionId transaction hash
+   * @param {number} blockHeight height of block which contain transaction
+   * @return {Promise<boolean>} indicate that custom transaction was affected on chain
+   */
+  async isAppliedCustomTransaction (transactionId: string, blockHeight: number): Promise<boolean> {
+    return await this.client.call('isappliedcustomtx', [transactionId, blockHeight], 'bignumber')
```
Comment (surangap) [inline packages/jellyfish-api-core/src/category/masternode.ts:209]: does not need `bignumber` precision.
```suggestion
    return await this.client.call('isappliedcustomtx', [transactionId, blockHeight], 'number')
```

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      tradeable: true,
+      collateralAddress: silverAddress
+    }
+    const silverTokenId = await client.token.createToken(silverMetadata)
+    await container.generate(1)
+    const silverHeight = await client.blockchain.getBlockCount()
+
+    const cupperAddress = await container.getNewAddress('', 'legacy')
+    const cupperMetadata = {
+      symbol: 'CUPPER',
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:46]: Copper?

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  })
+
+  it('should be failed while using invalid id', async () => {
+    const blockHeight = await client.blockchain.getBlockCount()
+
+    const result = await client.masternode.isAppliedCustomTransaction('b2bb09ffe9f9b292f13d23bafa1225ef26d0b9906da7af194c5738b63839b235', blockHeight)
+    expect(result).toStrictEqual(false)
+
+    await expect(client.masternode.isAppliedCustomTransaction('z2bb09ffe9f9b292f13d23bafa1225ef26d0b9906da7af194c5738b63839b235', blockHeight)).rejects.toThrow()
+  })
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:92]: add tests to cover conditionals in https://github.com/DeFiCh/ain/blob/7630137b3ea10269f564f13297f21b997735f38e/src/rpc/util.cpp#L84 for completeness. Also in the test, check for at least a part of the error string.

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:5]: try to use `testing.container`, we plan to put the container creation inside `testing` class.
you can find sample code in loan tests.
@chee-chyuan FYI

**Review by surangap — CHANGES_REQUESTED** [review]
added some comments.

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import { Testing } from '@defichain/jellyfish-testing'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+  const client = new ContainerAdapterClient(container)
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:8]: inside testing, you already have a RPC client. can use that

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      symbol: 'GOLD',
+      name: 'shiny gold',
+      isDAT: false,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: goldAddress
+    }
+    const goldTokenId = await client.token.createToken(goldMetadata)
+    await testing.container.generate(1)
+    const goldHeight = await client.blockchain.getBlockCount()
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:31]:
```suggestion
    const goldHeight = await testing.rpc.blockchain.getBlockCount()
```

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
+import { Testing } from '@defichain/jellyfish-testing'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+
+describe('Masternode', () => {
+  const testing = Testing.create(new MasterNodeRegTestContainer())
+  const container = testing.container
+  const client = testing.rpc
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:7]:
```suggestion
```

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:11]:
```suggestion
        await testing.container.start()
        await testing.container.waitForWalletCoinbaseMaturity()
```

File: packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+  })
+
+  afterAll(async () => {
+    await container.stop()
```
Comment (surangap) [inline packages/jellyfish-api-core/__tests__/category/masternode/isAppliedCustomTransaction.test.ts:15]:
```suggestion
    await testing.container.stop()
```

**Review by canonbrother — DISMISSED** [review]

**Review by surangap — APPROVED** [review]

**Review by canonbrother — APPROVED** [review]

**Review by chee-chyuan — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]

**Pull request merged**